src/OpenXRApi.cpp

```diff
@@ -1208,6 +1339,42 @@ void OpenXRApi::update_controllers() {
 	};
 }
 
+void OpenXRApi::update_handtracking() {
+	const XrTime time = frameState.predictedDisplayTime;
+	XrResult result;
+
+	if (!hand_tracking_supported) {
+		return;
+	}
+
+	for (int i = 0; i < HANDCOUNT; i++) {
+		XrHandJointsLocateInfoEXT locateInfo = {
+			.type = XR_TYPE_HAND_JOINTS_LOCATE_INFO_EXT,
+			.baseSpace = play_space,
+			.time = time,
+		};
+
+		result = xrLocateHandJointsEXT(hand_trackers[i].hand_tracker, &locateInfo, &hand_trackers[i].locations);
+		if (xr_result(result, "failed to get tracking for hand %d!", i)) {
+			// For some reason an inactive controller isn't coming back as inactive but has coordinates either as NAN or very large
+			const XrPosef &palm = hand_trackers[i].joint_locations[XR_HAND_JOINT_PALM_EXT].pose;
+			if (
+				!hand_trackers[i].locations.isActive
+				|| isnan(palm.position.x)
```

Comment:
> If the returned isActive is true, the runtime must return all joint locations with both XR_SPACE_LOCATION_POSITION_VALID_BIT and XR_SPACE_LOCATION_ORIENTATION_VALID_BIT set.

Pretty sure the position can't be nan and `valid` at the same time.

Reply:
Yup, that's how it should be, unfortunately that is not how it is, I'm getting junk values when the trackers aren't tracking yet while isActive is true. It's not even NAN all the time, sometimes its just really really really large numbers. 
Either there is something else I need to check to see if the tracking data is valid yet or there is a bug in Valves implementation.